proxy/proxy-tests/src/test/scala/io/cloudstate/proxy/eventing/GooglePubsubSpec.scala

```diff
@@ -0,0 +1,133 @@
+    locateBaseDir(new File("."))
+  }
+  val nodeSupportDir = new File(baseDir, "node_support")
+  val crdtExampleDir = new File(new File(baseDir, "samples"), "js-shopping-cart")
+
+  // We only need to start one user function, since the multiple Akka nodes can all connect to it, it doesn't make
+  // a difference whether they use different services or the same.
+  var userFunction: Process = _
+  var node: ActorSystem = _
+  var testSystem: ActorSystem = _
+  var support: GCPubsubEventingSupport = _
+  var materializer: ActorMaterializer = _
+
+  override implicit def patienceConfig: PatienceConfig = PatienceConfig(Span(20, Seconds), Span(1, Second))
+
+  "Google Pubsub Eventing support" should {
+    implicit def implicitTestSystem = testSystem
+    implicit def implicitMaterializer = materializer
+    implicit def implicitDispatcher = implicitTestSystem.dispatcher
+    "be tested" in pending // FIXME ADD TESTS HERE
+  }
+
+  override protected def beforeAll(): Unit = {
+    val projectId = "test"
+    val pubsub_port = 8085
+    val pubsub_host = "localhost"
+
+    implicit val s = ActorSystem(
+      "GooglePubsubSpec",
+      ConfigFactory.parseString("""
+        |akka.remote.netty.tcp.port = 0
+        |akka.remote.artery.canonical.port = 0
+        |akka.remote.artery.bind.port = 0
+        |akka.coordinated-shutdown.exit-jvm = off
+        |akka.jvm-exit-on-fatal-error = false
+      """.stripMargin).withFallback(ConfigFactory.defaultReference()).resolve()
+    )
+    implicit val m = ActorMaterializer()
+    implicit val d = s.dispatcher
+
+    val pubsub =
+      EventingManager
+        .createSupport(ConfigFactory.parseString(s"""
+        |support = "google-pubsub"
+        |google-pubsub.host = "${pubsub_host}"
+        |google-pubsub.port = ${pubsub_port}
+        |google-pubsub.rootCa = "none"
+        |google-pubsub.callCredentials = "none"
+        |google-pubsub.project-id = "${projectId}"
+        |google-pubsub.poll-interval = 1s
+        |google-pubsub.upstream-ack-deadline = 10s
+        |google-pubsub.downstream-batch-deadline = 5s
+        |google-pubsub.downstream-batch-size = 10
+        |google-pubsub.manage-topics-and-subscriptions = "by-proxy"
+      """.stripMargin))
+        .collect({ case g: GCPubsubEventingSupport => g })
+        .getOrElse(throw new IllegalStateException("Unable to create EventingSupport for Google Pubsub"))
+
+    testSystem = s
+    materializer = m
+    support = pubsub
+
+    userFunction = Process(Seq("node", "index.js"), crdtExampleDir).run()
+
+    node = CloudStateProxyMain.start(
+      ConfigFactory.load("dev-mode.conf").withFallback(ConfigFactory.defaultReference()).resolve()
+    )
+
+    akka.serialization.JavaSerializer.currentSystem.withValue(testSystem.asInstanceOf[ExtendedActorSystem]) {
+      import akka.pattern.ask
+      implicit val timeout = Timeout(1.seconds)
+      val manager = node.actorSelection(node / "server-manager-supervisor" / "server-manager")
```

Comment:
- Would this result, when invoked (downstream), in the restart of the subordinate, along with clearing out its accumulated state?

Reply:
This is a test case, and the only thing we do here is to make sure that the server eventually becomes ready to accept requests.